WordPress/Classes/ViewRelated/Me/App Settings/Boolean User Defaults/BooleanUserDefaultsDebugViewModel.swift

```diff
@@ -73,22 +84,28 @@ final class BooleanUserDefaultsDebugViewModel: ObservableObject {
         return filteredSections
     }
 
-    private func processGroupedUserDefaults(_ userDefaults: [String: Bool]) -> [Row] {
-        var rows = userDefaults.reduce(into: [Row]()) { result, keyValue in
-            let (key, value) = keyValue
-            result.append(processSingleUserDefault(key: key, value: value))
+    private func processGroupedUserDefaults(_ userDefaults: [String: Bool]) async -> [Row] {
+        let rows = try? await self.coreDataStack.performAndSave { [weak self] context -> [Row] in
```

Comment:
When fetching objects from Core Data from a background thread, we cannot use the `mainContext` as it should be only used in the main thread. The app might crash if the main context is called on a background thread.

The method `coreDataStack.performAndSave` created a new context, and executes the block in the background thread. Then, we use that context to fetch objects from Core Data.